### PR TITLE
Edtran/security Fix eslint

### PR DESCRIPTION
**Pull request by xTEddie**

### CODE REVIEW

File: src/core/IC3Initializer.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (grantborthwick) [inline src/core/IC3Initializer.ts:1]: Should we disable this? Could we use unknown instead?

**Review by grantborthwick — APPROVED** [review]

File: src/common/Utilities.ts [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
         return Utilities.isNullOrUndefined(s) || s === "";
     }
 
-    public static isNullOrUndefinedOrEmptyArray(obj: any[]) {
+    public static isNullOrUndefinedOrEmptyArray(obj: any[]): boolean {
         return Utilities.isNullOrUndefined(obj) || obj.length === 0;
     }
 
-    public static isNullOrUndefined(obj: any) {
+    public static isNullOrUndefined(obj: any): boolean { // eslint-disable-line
```
Comment (shurane) [inline src/common/Utilities.ts:15]: Should these be more specific instead of `eslint-disable-line`?

File: src/common/Utilities.ts [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
         return (obj === null || obj === undefined);
     }
 
-    public static isPrimitiveString(obj: any) {
+    public static isPrimitiveString(obj: any): boolean { // eslint-disable-line
```
Comment (shurane) [inline src/common/Utilities.ts:19]: Same as above

**Review by shurane — APPROVED** [review]

File: src/common/Util.ts [hunk 4]
```diff
@@ -151,7 +155,7 @@ export default class Util {
         let name = '';
         let extension = '';
         if (nameAndExtensionMatcher.length > 1) {
-            extension = nameAndExtensionMatcher.pop()!.toLowerCase();
+            extension = nameAndExtensionMatcher.pop()!.toLowerCase(); // eslint-disable-line @typescript-eslint/no-non-null-assertion
```
Comment (shurane) [inline src/common/Util.ts:158]: Since there's a `length > 1`, can the `!` non-null assertion operator be removed?

File: src/common/Utilities.ts [hunk 5]
```diff
@@ -41,8 +43,8 @@ export default class Utilities {
                 finalPath += "?";
             }
             for (const queryParamKey in queryParameters) {
-                if (queryParameters.hasOwnProperty(queryParamKey)) {
-                    let queryParamData = queryParamKey + "=" + queryParameters[queryParamKey];
+                if (queryParameters.hasOwnProperty(queryParamKey)) { // eslint-disable-line no-prototype-builtins
+                    let queryParamData = queryParamKey + "=" + queryParameters[queryParamKey]; // eslint-disable-line security/detect-object-injection
```
Comment (shurane) [inline src/common/Utilities.ts:47]: Could we change this to to use a template string, a la `${queryParamKey}=${queryParameters[queryParamKey]}`?

Comment (shurane) [reply src/common/Utilities.ts:47]: Actually nevermind.

File: src/common/Util.ts (hunk 4 above)
Comment (xTEddie) [reply src/common/Util.ts:158]: Typescript would complain that `Object is possibly 'undefined'`

**Review by shengchenmicrosoftworkemail — APPROVED** [review]

File: src/core/IC3Initializer.ts (hunk 1 above)
Comment (xTEddie) [reply src/core/IC3Initializer.ts]: Good point. A good portion of it is fixed. There's still some, but they're not as simple as just changing the typing on one file.

**Pull request merged**